src/features/nav/SideNav.tsx

```diff
@@ -0,0 +1,134 @@
+import { Suspense, useEffect, useState } from 'react';
+
+import { HomeIcon } from '@radix-ui/react-icons';
+import { QueryClient } from 'react-query';
+import { useLocation } from 'react-router-dom';
+
+import { getCircleFromPath, getOrgFromPath, paths } from '../../routes/paths';
+import { Flex } from '../../ui';
+
+import { NavCircle, NavOrg, QUERY_KEY_NAV, useNavQuery } from './getNavData';
+import { NavCircles } from './NavCircles';
+import { NavClaimsButton } from './NavClaimsButton';
+import { NavItem } from './NavItem';
+import { NavLogo } from './NavLogo';
+import { NavOrgs } from './NavOrgs';
+import { NavProfile } from './NavProfile';
+
+export const InvalidateSideNav = async (queryClient: QueryClient) => {
+  await queryClient.invalidateQueries(QUERY_KEY_NAV);
+};
+
+export const SideNav = () => {
+  /*
+    TODO: review semantic color names
+    TODO: icons need fixing
+    TODO: a11y
+    TODO: favicon
+    TODO: scrolly gradient
+    TODO: what's your name prompt
+   */
+  const [currentCircle, setCurrentCircle] = useState<NavCircle | undefined>(
+    undefined
+  );
+  const [currentOrg, setCurrentOrg] = useState<NavOrg | undefined>(undefined);
+
+  const location = useLocation();
+  const { data } = useNavQuery();
+
+  const showClaimsButton = (data?.claims_aggregate.aggregate?.count || 0) > 0;
+
+  const setCircleAndOrgIfMatch = (orgs: NavOrg[]) => {
+    const circleId = getCircleFromPath(location);
+    const orgId = getOrgFromPath(location);
+
+    for (const o of orgs) {
+      for (const c of o.circles) {
```

Comment:
This is a bit weird to me because we sometimes have the circleId but NOT the orgId, but not sure a way to improve it...

Reply:
maybe a tiny improvement